packages/material-ui/src/Switch/Switch.js

```diff
@@ -121,11 +123,37 @@ export const styles = theme => ({
       theme.palette.type === 'light' ? theme.palette.common.black : theme.palette.common.white,
     opacity: theme.palette.type === 'light' ? 0.38 : 0.3,
   },
+  /* Small size */
+  sizeSmall: {
+    width: 30 + 10 * 2,
+    height: 10 + 10 * 2,
+    padding: 10,
+  },
+  thumbSmall: {
+    width: 18,
+    height: 18,
+  },
+  switchBaseSizeSmall: {
+    padding: 6,
+  },
 });
 
 const Switch = React.forwardRef(function Switch(props, ref) {
-  const { classes, className, color = 'secondary', edge = false, ...other } = props;
-  const icon = <span className={classes.thumb} />;
+  const {
+    classes,
+    className,
+    color = 'secondary',
+    edge = false,
+    size = 'medium',
+    ...other
+  } = props;
+  const icon = (
+    <span
+      className={clsx(classes.thumb, {
+        [classes[`thumb${capitalize(size)}`]]: size !== 'medium',
```

Comment:
What do you think of increasing specificity to target this deep element from the root?